drivers/sensor/dht/dht.c

```diff
@@ -215,23 +220,30 @@ static const struct sensor_driver_api dht_api = {
 
 static int dht_init(struct device *dev)
 {
+	int rc = 0;
 	struct dht_data *drv_data = dev->driver_data;
+	const struct dht_config *cfg = dev->config->config_info;
 
-	drv_data->gpio = device_get_binding(CONFIG_DHT_GPIO_DEV_NAME);
+	drv_data->gpio = device_get_binding(cfg->ctrl);
 	if (drv_data->gpio == NULL) {
-		LOG_ERR("Failed to get GPIO device.");
+		LOG_ERR("Failed to get GPIO device %s.", cfg->ctrl);
 		return -EINVAL;
 	}
 
-	gpio_pin_configure(drv_data->gpio, CONFIG_DHT_GPIO_PIN_NUM,
-			   GPIO_DIR_OUT);
-
-	gpio_pin_write(drv_data->gpio, CONFIG_DHT_GPIO_PIN_NUM, 1);
+	rc = gpio_pin_configure(drv_data->gpio, cfg->pin, GPIO_DIR_OUT);
+	if (rc == 0) {
+		rc = gpio_pin_write(drv_data->gpio, cfg->pin, 1);
+	}
 
-	return 0;
+	return rc;
 }
 
-struct dht_data dht_data;
+static struct dht_data dht_data;
+static struct dht_config dht_config = {
```

Comment:
this should be `const`

Reply:
done; thanks.

Reply:
you amended the wrong commit. the change should be in `sensor: dht: convert from Kconfig to devicetree`

Reply:
Moved to the right commit.